### PR TITLE
contracts refactor, added comments

### PR DESCRIPTION
**Pull request by ordd**
# Description

add more comments and make sure they are correct.
make sure to adhere to recommended code syntax in quantstamp original report.

About # (link your issue here)
#132 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: stakingModel/contracts/FirstClaimPool.sol [hunk 1]
```diff
@@ -14,7 +14,11 @@ import "./UBIScheme.sol";
 contract FirstClaimPool is FeelessScheme, ActivePeriod {
     using SafeMath for uint256;
 
+    // The whitelisted ubi scheme contrat
```
Comment (sirpy) [inline stakingModel/contracts/FirstClaimPool.sol]:
```suggestion
    // The whitelisted ubi scheme contract
```

File: stakingModel/contracts/GoodMarketMaker.sol [hunk 2]
```diff
@@ -288,16 +351,19 @@ contract GoodMarketMaker is BancorFormula, DSMath, SchemeGuard {
...
+    * @return price of GD
+    */
     function currentPrice(ERC20 _token)
         public
         view
         onlyActiveToken(_token)
         returns (uint256)
     {
         ReserveToken memory rtoken = reserveTokens[address(_token)];
+        require(rtoken.gdSupply > 0, "Reserve token not initialized");
```
Comment (sirpy) [inline stakingModel/contracts/GoodMarketMaker.sol]: why do we need this?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: stakingModel/contracts/FirstClaimPool.sol (hunk 1 above)
Comment (ordd) [reply stakingModel/contracts/FirstClaimPool.sol]: 181bcd8c3bab83505b8c2e181fa3b1a121496f95

File: stakingModel/contracts/GoodMarketMaker.sol (hunk 2 above)
Comment (ordd) [reply stakingModel/contracts/GoodMarketMaker.sol]: removed
181bcd8c3bab83505b8c2e181fa3b1a121496f95

File: stakingModel/contracts/GoodReserveCDai.sol [hunk 3]
```diff
@@ -208,6 +202,7 @@ contract GoodReserveCDai is DSMath, FeelessScheme, ActivePeriod {
      */
     function start() public onlyRegistered {
         addRights();
+        GoodDollar gooddollar = GoodDollar(address(avatar.nativeToken()));
```
Comment (sirpy) [inline stakingModel/contracts/GoodReserveCDai.sol]: why do we need this if you read it dynamically eveyrwhere?

Comment (ordd) [reply stakingModel/contracts/GoodReserveCDai.sol]: we need to add the reserve as a minter
updated 53102ee50aa12ced8bb5ae3e39d40ea62ded976d

**Pull request merged**